regression in 5.2? cannot extract `rest_framework` `Response` from django test client

# Bug report


seems something in 5.2 caused this to break -- here's a minimal example:

```python
from django.test import TestCase
from rest_framework.response import Response


class T(TestCase):
    def get_rf_resp(self, url: str) -> Response:
        resp = self.client.get(url)
        assert isinstance(resp, Response), resp
        return resp
```


## What's wrong


```console
$ mypy t3.py --warn-unreachable
t3.py:8: error: Subclass of "_MonkeyPatchedWSGIResponse" and "Response" cannot exist: would have incompatible method signatures  [unreachable]
t3.py:9: error: Statement is unreachable  [unreachable]
Found 2 errors in 1 file (checked 1 source file)
```

## How is that should be

no errors

## System information

- OS:
- `python` version: 3.13.1
- `django` version: 5.2.0
- `mypy` version: 1.15.0
- `django-stubs` version: 5.2.0 (forked, but not relevant here)
- `django-stubs-ext` version: 5.2.0

5.2.1 I have a similar problem now with StreamingHttpResponse:

```
tests/sentry/api/test_base.py:495: error: Subclass of "Response" and "StreamingHttpResponse" cannot exist: would have incompatible method signatures  [unreachable]
```

--- a/django-stubs/test/client.pyi
+++ b/django-stubs/test/client.pyi
@@ -223,7 +223,8 @@ class _MonkeyPatchedWSGIResponse(_WSGIResponse):
     # `HTTPResponse` when API failed and `TemplateResponse` when successful.
     # https://docs.djangoproject.com/en/stable/topics/testing/tools/#django.test.Response.context
     context: ContextList | dict[str, Any]
-    context_data: ContextList | dict[str, Any]
+    # Must match `django.template.response.SimpleTemplateResponse.context_data`
+    context_data: dict[str, Any] | None
 
     content: bytes
     resolver_match: ResolverMatch
@@ -244,7 +245,8 @@ class _MonkeyPatchedASGIResponse(_ASGIResponse):
     # `HTTPResponse` when API failed and `TemplateResponse` when successful.
     # https://docs.djangoproject.com/en/stable/topics/testing/tools/#django.test.Response.context
     context: ContextList | dict[str, Any]
-    context_data: ContextList | dict[str, Any]
+    # Must match `django.template.response.SimpleTemplateResponse.context_data`
+    context_data: dict[str, Any] | None
 
     content: bytes
     resolver_match: ResolverMatch

--- a/tests/typecheck/test/test_client.yml
+++ b/tests/typecheck/test/test_client.yml
@@ -8,7 +8,7 @@
       reveal_type(response.templates)  # N: Revealed type is "builtins.list[django.template.base.Template]"
       reveal_type(response.client)  # N: Revealed type is "django.test.client.Client"
       reveal_type(response.context)  # N: Revealed type is "Union[django.test.utils.ContextList, builtins.dict[builtins.str, Any]]"
-      reveal_type(response.context_data)  # N: Revealed type is "Union[django.test.utils.ContextList, builtins.dict[builtins.str, Any]]"
+      reveal_type(response.context_data)  # N: Revealed type is "Union[builtins.dict[builtins.str, Any], None]"
       reveal_type(response.content)  # N: Revealed type is "builtins.bytes"
       reveal_type(response.redirect_chain)  # N: Revealed type is "builtins.list[Tuple[builtins.str, builtins.int]]"
       reveal_type(response.text)  # N: Revealed type is "builtins.str"
@@ -24,7 +24,7 @@
         reveal_type(response.templates)  # N: Revealed type is "builtins.list[django.template.base.Template]"
         reveal_type(response.client)  # N: Revealed type is "django.test.client.AsyncClient"
         reveal_type(response.context)  # N: Revealed type is "Union[django.test.utils.ContextList, builtins.dict[builtins.str, Any]]"
-        reveal_type(response.context_data)  # N: Revealed type is "Union[django.test.utils.ContextList, builtins.dict[builtins.str, Any]]"
+        reveal_type(response.context_data)  # N: Revealed type is "Union[builtins.dict[builtins.str, Any], None]"
         reveal_type(response.content)  # N: Revealed type is "builtins.bytes"
         reveal_type(response.redirect_chain)  # N: Revealed type is "builtins.list[Tuple[builtins.str, builtins.int]]"
         response.json()

